### ISSUE TITLE
Empty Tuple Method

### ISSUE THREAD

**Issue by ParadaCarleton** [JuliaArrays/FillArrays.jl#167]
When `Zeros` or `Ones` gets an empty tuple as its only argument, perhaps we should return `zero` or `one`? This way functions that can be called on both arrays and numbers can be simplified, instead of having to special-case numbers.

**Comment by dlfivefifty** [comment]
I don’t understand. Shouldn’t it be a 0-dimensional ` Zeros ` / ` Ones `? 

One should never return a different type from a type constructor

**Comment by ParadaCarleton** [comment]
That's true. Maybe a `lazy_zero` method with this behavior? (i.e. return `Zeros` for array arguments and `0` for number-like arguments.)
